auditbeat/module/auditd/ecs_categorization.yml

```diff
@@ -0,0 +1,320 @@
+    # AUDIT_VIRT_RESOURCE - Resource assignment
+    # AUDIT_VIRT_MACHINE_ID - Binding of label to VM
+    # AUDIT_VIRT_INTEGRITY_CHECK - Guest integrity results
+    # AUDIT_VIRT_CREATE - Creation of guest image
+    # AUDIT_VIRT_DESTROY - Destruction of guest image
+    # AUDIT_VIRT_MIGRATE_IN - Inbound guest migration info
+    # AUDIT_VIRT_MIGRATE_OUT - Outbound guest migration info
+    categories:
+      - host
+    default_types:
+      - info
+
+  EventTypeUserspace:
+    # AUDIT_CHGRP_ID - User space group ID changed
+    # AUDIT_TEST - Used for test success messages
+    # AUDIT_TRUSTED_APP - Trusted app msg - freestyle text
+    # AUDIT_USER_CMD - User shell command and args
+    # AUDIT_CHUSER_ID - Changed user ID supplemental data
+    categories:
+      - process
+    default_types:
+      - info
+    types:
+      AUDIT_CHUSER_ID:
+        - change
+      AUDIT_CHGRP_ID:
+        - change
+
+  EventTypeSystemServices:
+    # AUDIT_KERNEL - Asynchronous audit record. NOT A REQUEST.
+    # AUDIT_SYSTEM_BOOT - System boot
+    # AUDIT_SYSTEM_SHUTDOWN - System shutdown
+    # AUDIT_SYSTEM_RUNLEVEL - System runlevel change
+    # AUDIT_SERVICE_START - Service (daemon) start
+    # AUDIT_SERVICE_STOP - Service (daemon) stop
+    categories:
+      - host
+    default_types:
+      - info
+    types:
+      AUDIT_SYSTEM_BOOT:
+        - start
+      AUDIT_SYSTEM_RUNLEVEL:
+        - change
+      AUDIT_SYSTEM_SHUTDOWN:
+        - end
+
+  EventTypeAuditDaemon:
+    # AUDIT_DAEMON_START - Daemon startup record
+    # AUDIT_DAEMON_END - Daemon normal stop record
+    # AUDIT_DAEMON_ABORT - Daemon error stop record
+    # AUDIT_DAEMON_CONFIG - Daemon config change
+    categories:
+      - process
+    default_types:
+      - info
+    types:
+      AUDIT_DAEMON_START:
+        - start
+      AUDIT_DAEMON_END:
+        - end
+
+  EventTypeDACDecision:
+    # AUDIT_SECCOMP - Secure Computing event
+    categories:
+      - process
+    default_types:
+      - change
+
+  EventTypeConfig:
+    # AUDIT_USYS_CONFIG - User space system config change
+    # AUDIT_CONFIG_CHANGE - Audit system configuration change
+    # AUDIT_NETFILTER_CFG - Netfilter chain modifications
+    # AUDIT_FEATURE_CHANGE - audit log listing feature changes
+    # AUDIT_REPLACE - Replace auditd if this packet unanswerd
+    categories:
+    #   - config
+
+    EventTypeTTY:
+    # AUDIT_USER_TTY - Non-ICANON TTY input meaning
+    # AUDIT_TTY - Input on an administrative TTY
+    categories:
+    #   - device
+
+  EventTypeMACDecision:
+    # AUDIT_USER_SELINUX_ERR - SE Linux user space error
+    # AUDIT_USER_AVC - User space avc message
+    # AUDIT_APPARMOR_ALLOWED
+    # AUDIT_APPARMOR_DENIED
+    # AUDIT_APPARMOR_ERROR
+    # AUDIT_AVC - SE Linux avc denial or grant
+    # AUDIT_SELINUX_ERR - Internal SE Linux Errors
+    # AUDIT_AVC_PATH - dentry, vfsmount pair from avc
+    categories:
+    #     - policy
+
+  EventTypeAnomoly:
```

Comment:
```suggestion
  EventTypeAnomaly:
```

Reply:
Some of the anomalies below sound like they could be associated with existing categories, such as "iam", "file", "process", "authentication" and so on.

Then if we later add an "anomaly" category, these events could also be assigned that category.

Reply:
yeah, that was my thought too--some of them are `policy` violations (like exceeding some threshold of failed logins), some are `device` access/modifications, and some are restricted `process` spawning, `file` or `iam` creation. So I'd be happy with categorizing them there.

Wondering though if it makes sense to add a secondary category in these cases--they're only being signaled because `auditd` has flagged them as "anomalous" - so they aren't your standard `file`/`creation` events, etc.?

Reply:
ah, just re-read your comment, you were thinking the exact same thing as me :)

Reply:
Yeah but it still brings a good point.

If this is a second "anomaly" event, will this mean this anomalous IAM action yield two auditd events? One for the regular IAM event, and this second one for the anomaly?

But I'd err on the side of capturing both as IAM, and then we / the user can dedupe if that's the case. 

Moreover, if a user turns off "normal" IAM events to reduce noise, this IAM anomaly should be categorized as IAM.